easybuild/tools/options.py

```diff
@@ -724,6 +730,31 @@ def postprocess(self):
             self.show_config()
             cleanup_and_exit(self.tmpdir)
 
+    def _postprocess_optarch(self):
+        """Postprocess --optarch option."""
+        optarch_parts = self.options.optarch.split(OPTARCH_SEP)
+        n_compilers = len(optarch_parts)
+        
+        # if the format is not adequate we refuse continuing
+        if n_compilers > 1 and n_compilers != self.options.optarch.count(OPTARCH_MAP_CHAR):
+            raise EasyBuildError("The optarch option has an incorrect syntax: %s" % self.options.optarch)
+        else:
+            logger = fancylogger.getLogger()
+            # if there are options for different compilers, we set up a dict
+            if OPTARCH_MAP_CHAR in optarch_parts[0]:
+                optarch_dict = {}
+                for compiler, compiler_opt in [x.split(OPTARCH_MAP_CHAR) for x in optarch_parts]:
+                    if compiler in optarch_dict:
+                        raise EasyBuildError("The optarch option contains duplicated entries for compiler %s: %s" 
```

Comment:
Same comment as above. I am ok with the change, but we should be consistent (to be done in a different PR?)